pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/JsonExtractScalarTransformFunction.java

```diff
@@ -184,8 +184,12 @@ public long[] transformToLongValuesSV(ValueBlock valueBlock) {
       if (result instanceof Number) {
         _longValuesSV[i] = ((Number) result).longValue();
       } else {
-        // Handle scientific notation
-        _longValuesSV[i] = (long) Double.parseDouble(result.toString());
+        try {
```

Comment:
@Jackie-Jiang @gortiz 
Do we need to support the notation ? 
If so, what do you think of copying parseLong code and using a static exception instead to curb potential allocations and stack trace `business`.

Reply:
Seems like we are mixing `CAST` into this method.
IMO the correct fix should be:
- Use `Long.parseLong()` here
- Check result type, and fallback to default when result type doesn't match method (see `CastTransformFunction` as an example)

Seems other json extract function also have this problem

Reply:
I think we could iterate on the string looking for numbers or -. If we only find them, we optimistically call `Long.parseLong`. If that fails or if we find anything different than `-` or numbers, we call `Double.parseDouble`. Unless I missing something this should work in almost all case except very large numbers (which we can probably also check statically by comparing the length of the string).

Reply:
IMO if user specifies it to extract as `LONG`, the value should be either a number, or a long value string.
I guess the reason why we parse it as double here is to handle the scenario where upper level directly asks for double value. E.g. `SUM(JSON_EXTRACT_SCALAR())`

Reply:
My suggestion would be:
1. Use `Long.parseLong()` without extra try-catch
2. Fall back to `BaseTransformFunction` if the result **stored** type doesn't match the function name (e.g. result stored type is `DOUBLE` but function name is `transformToLongValuesSV`)

To demonstrate the second point, you can try calling `transformToLongValuesSV()` on `JSON_EXTRACT_SCALAR(..., 'DOUBLE')` which might fail if we only do 1 without 2